docker/docker_client.go

```diff
@@ -32,6 +32,11 @@ const (
 	blobUploadURL = "%s/blobs/uploads/"
 )
 
+// Client is a generic client interface
+type Client interface {
+	MakeRequest(string, string, map[string][]string, io.Reader, bool) (*http.Response, error)
+}
+
```

Comment:
Why export this at all? Do you expect any external users? If this is only for testing, we can make that interface private, and hide its uses from clients (see below).